### PR TITLE
fix(#121): handle join_txn_mode for version_session

### PR DESCRIPTION
**Pull request by ashish16052**
- SQLA 2.0 introduced new session param called `join_transaction_mode` which changes the behaviour of how existing transactions are handled

- When using a Connection with an existing SAVEPOINT: `SQLA-1.4`: the Session would propagate both commit() and rollback() calls to the underlying transaction. `SQLA-2.0`: use join_transaction_mode=control_fully to replicate the same behaviour.

- When using a Connection without an existing SAVEPOINT: `SQLA-1.4`: the Session would propagate only rollback() calls to the underlying transaction. `SQLA-2.0`: use join_transaction_mode=rollback_only to replicate the same behaviour.

- use `rollback_only` mode while creation of version_session, as it is using an existing connection, and we are never doing a commit() on it.

Documentation:[ join_transaction_mode](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.params.join_transaction_mode)

### CODE REVIEW

File: sqlalchemy_history/unit_of_work.py [hunk 1]
```diff
@@ -21,6 +21,9 @@ def __init__(self, manager):
         self.manager = manager
         self.reset()
 
+    def create_version_session(self, session):
+        return sa.orm.session.Session(bind=session.connection(), join_transaction_mode="rollback_only")
```
Comment (indiVar0508) [inline sqlalchemy_history/unit_of_work.py:25]: We would also have to handle sqla 1.4 here aswell.

current CI workflow seems to have a bug, where sqla-1.4 is not getting installed in poetry's env, 

can you also fix that?, if the fix breaks other items in CI we can create a seperate issue for it.

File: sqlalchemy_history/unit_of_work.py [hunk 2]
```diff
@@ -65,7 +68,7 @@ def process_before_flush(self, session):
             return
 
         if not self.version_session:
-            self.version_session = sa.orm.session.Session(bind=session.connection())
+            self.version_session = self.create_version_session(session)
```
Comment (indiVar0508) [inline sqlalchemy_history/unit_of_work.py:71]: can you also add testcase for this behaviour.

File: sqlalchemy_history/unit_of_work.py (hunk 1 above)
Comment (ashish16052) [reply sqlalchemy_history/unit_of_work.py]: Tried fixing this by replacing:
```pip install 'sqlalchemy${{ matrix.sqla }}'```
 with
```poetry add "sqlalchemy${{ matrix.sqla }}"```

it is breaking other tests

File: sqlalchemy_history/unit_of_work.py [hunk 3]
```diff
@@ -21,6 +21,12 @@ def __init__(self, manager):
         self.manager = manager
         self.reset()
 
+    def create_version_session(self, session):
+        try:
```
Comment (indiVar0508) [inline sqlalchemy_history/unit_of_work.py:25]: Test seems to be working with rebased code, 
can you add small note saying what happened here so if someone comes cold to this code and has no gitlens is not confused.

**Review by indiVar0508 — APPROVED** [review]
add a minor comment

**Pull request merged**